cli/mender-artifact/write.go

```diff
@@ -25,8 +25,45 @@ import (
 	"github.com/mendersoftware/mender-artifact/handlers"
 	"github.com/pkg/errors"
 	"github.com/urfave/cli"
+	"io"
+	"io/ioutil"
 )
 
+func writeRootfsImageChecksum(c *cli.Context, typeInfo *artifact.TypeInfoV3) (err error) {
+	chk := artifact.NewWriterChecksum(ioutil.Discard)
+	payload, err := os.Open(c.String("file"))
+	if err != nil {
+		return cli.NewExitError(fmt.Sprintf("Failed to open the payload file: %q", c.String("file")), 1)
+	}
+	if _, err = io.Copy(chk, payload); err != nil {
+		return cli.NewExitError("Failed to generate the checksum for the payload", 1)
+	}
+	checksum := string(chk.Checksum())
+
+	Log.Debugf("Adding the `rootfs_image_checksum`: %q to Artifact provides", checksum)
+	if typeInfo == nil {
+		return errors.New("Type-info is unitialized")
+	}
+	if typeInfo.ArtifactProvides == nil {
```

Comment:
@kacf this is quite possibly the fugliest this in our code-base yet. I'm ashamed

Reply:
Magic is more than welcome here. Muggle hacking also for that matter.

Reply:
Ignore this fugliness, I got the shits and did: https://github.com/mendersoftware/mender-artifact/pull/246, so rebasing after it is merged.